File: src/main/java/biblioteca/RecursoDigital.java
package biblioteca;

public class RecursoDigital extends Material {
    private String url;
    private int tamanioMb;
    private int accesosSimultaneos = 0;
    private static final int MAX_ACCESOS = 5;

    public RecursoDigital(String url, int tamanioMb) {
    this.url = url;
    this.tamanioMb = tamanioMb;
}


    @Override
    public String prestar(String dniLector, int dias) {
        // Los recursos digitales permiten accesos simultáneos
        if (accesosSimultaneos >= MAX_ACCESOS)
            throw new IllegalStateException("Capacidad máxima");
        // No cambia estado a PRESTADO: permanece DISPONIBLE
        accesosSimultaneos++;
        return "Acceso digital habilitado para " + dniLector;
    }


    @Override
    public void devolver() {
        if (accesosSimultaneos > 0) accesosSimultaneos--;
    }
}

File: biblioteca/recurso_digital.py
# biblioteca/recurso_digital.py

from biblioteca.material import Material
from biblioteca.libro import EstadoMaterial


class RecursoDigital(Material):
    MAX_ACCESOS = 5

    def __init__(self, url: str, tamanio_mb: int,
                 titulo: str, autor: str, anio: int):
        super().__init__(titulo, autor, anio)  # ← valida el año
        self._url = url
        self._tamanio_mb = tamanio_mb
        self._accesos_simultaneos = 0

    def prestar(self, dni_lector: str, dias: int) -> str:
        if self._accesos_simultaneos >= self.MAX_ACCESOS:
            raise RuntimeError("Capacidad máxima")
        self._accesos_simultaneos += 1
        return f'Acceso digital habilitado para {dni_lector}'

    def devolver(self) -> None:
        if self._accesos_simultaneos > 0:
            self._accesos_simultaneos -= 1